src/librbd/ImageWatcher.cc

```diff
@@ -340,9 +340,10 @@ void ImageWatcher<I>::notify_header_update(librados::IoCtx &io_ctx,
 }
 
 template <typename I>
-void ImageWatcher<I>::notify_quiesce(uint64_t request_id,
```

Comment:
Nit: just wondering the thinking behind this change? Couldn't you just expose the `Operations::reserve_async_request_id`` and let `api/Group::snap_create` allocate an id for the quiesce request? It shouldn't matter if the snapshot create RPC generates a new request id, right?

Reply:
The code just seems cleaner to me this way.

When initially I was implementing`notify_quiesce` I was thinking that the source for request_id could be any monotonically increasing sequence. That is why I decided it should be provided to `ImageWatcher::notify_quiesce`  "externally". And at that time I chose  `Operations::m_async_request_seq` as the source just because it was convenient.  But later when the decision was made to support resending quiesce notifications and we started to use `ImageWatcher::m_async_pending` for tracking them, it restricted the possible source of Ids to `Operations::m_async_request_seq` only. And as this knowledge and use is concentrated in `ImageWatcher` it seems natural to me now that the `ImageWatcher` should reserve the Id itself.

Reply:
I guess I can't remember the original reason why `m_async_request_seq` was in `Operations` instead of in `ImageWatcher` (perhaps where it more rightfully belongs) and instead of `Operations` use a `ImageWatcher::reserve_async_request_id`-like method. 

I don't think it's really necessary to change it, but to me this change just highlighted the ownership oddity to me.

Reply:
I like the idea of having `m_async_request_seq` in `ImageWatcher`, probably not in this PR though.